README.md

```diff
@@ -17,6 +17,17 @@ A lot of the setup uses docker to encapsulate some of the dependencies, networki
 
 ### Testing the API
 1. Initialize the database and run the development server using `yarn run api:start:test`. This will delete what is currently in the database and replace it with fixtures for testing.
-2. In another terminal, run `yarn run test:api`.
+    
+    > You can also do all of this with 1 line of commands, using `yarn api:start:lazy`. This will spin up the db, the mock_auth server, and the api service.
+2. In another terminal, run `yarn test:api`.
 3. When done, use `yarn stop` or `docker-compose down` to shut everything down.
 
+### Adding Models
+Models are added by using Neogma's typed ModelFactory function. To make a new model:
+1. Create a model `.ts` file in the `http/api/lib/models` folder.
+2. Define the properties, the related node type (with ModelRelatedNodesI type), then define the instance & neogma model by using `NeogmaInstance` and `NeogmaModel` type. 
+    > Note that in order for Neogma to recognize the model, the property declared in the ModelRelatedNodeI type has to match the model label string.
+3. Add it to the `index.ts` file and add the model to the `allModels` type.
+4. Define a init**ModelName**Model function that returns the Neogma ModelFactory value. Include the db as it requires a valid session when creating the model, and add any dependencies (models that it has relations to) as a parameter.
+    > If you have a circular dependency issue (2 models rely on each other to initilialize) use the `.addRelationship({...})` method to add it later in the `initAllModels()` function.
+5. Use the function you have just created in the initAllModels function. Add it to the return value. 
```

Comment:
It's a little different from being re-declared, because we use the typings like a checklist. 

When we instantiate the model using the `ModelFactory` it's a place where we can ask Neogma to add more validators. The bare-minimum is `required: boolean`. If desired, we can add other validators such as `must be dividable by ... number` or  `min` and `max` values (not exhaustive, there's more). It looks like we are copy and pasting information, but it more so that we haven't defined any extra validators. 

The process of adding the model to the type `AllModels` is there because we need a generic parameter that has all of the models for the db query `db.exec(proc: (session: Session, all: AllModels) => DatabaseResponse)` which is used in queries like `db.exec(api.createUser(id));` and `await db.exec(api.createMeasurement(me));`. These queries requires access to different models.